libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2190,7 +2190,11 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 	for (size_t i = 0; i < numParameters; i++)
 	{
 		Type const& argType = *type(*callArguments[i]);
-		BoolResult result = argType.isImplicitlyConvertibleTo(*functionPointerType->parameterTypes()[i]);
+		BoolResult result = argType.isImplicitlyConvertibleTo(
+								*functionPointerType->asExternallyCallableFunction(
+									functionPointerType->kind() == FunctionType::Kind::DelegateCall
```

Comment:
Looking at the code above: this is always false, isn't it :-)? ``functionPointerType->kind() == FunctionType::Kind::DelegateCall`` is an error case above that returns early, right?

Reply:
Yes. You're right. Will change it in the next push.